src/twisted/web/http.py

```diff
@@ -1211,7 +1210,7 @@ def write(self, data):
                 and self.method != b"HEAD"
                 and self.code not in NO_BODY_CODES
             ):
-                headers.append((b"Transfer-Encoding", b"chunked"))
```

Comment:
This change in design means all headers end up in `responseHeaders`, which is perhaps a semantics change?

Reply:
Thanks for the update.

I don't remember reading it, but I guess that for Twisted API, the `transfer-encoding` and `set-cookie` headers are kind of a private headers, as we have separate methods to set them.

I was thinking that at some point Twisted API would reject setting the `tranfer-encoding` or `set-cookie` headers , and this is why we have the separate headers here... but I could not find any ticket about this.

------

I guess that it would be nice to raise a warning is a resource / user tries to set the transfer-encoding header.... but this is something for another issue/PR